### PR TITLE
feat: Filter product by country to get answered questions

### PR DESCRIPTION
**Pull request by Jagrutiti**
### What
- Added `country `parameter to the `UnansweredQuestionCollection` API.

### Closes
- #901

### CODE REVIEW

File: robotoff/app/api.py [hunk 1]
```diff
@@ -1150,13 +1150,19 @@ def on_get(self, req: falcon.Request, resp: falcon.Response):
...
         value_tag: str = req.get_param("value_tag")
+        country: Optional[str] = req.get_param("country")
         server_domain: Optional[str] = req.get_param("server_domain")
 
+        import pdb
+
+        pdb.set_trace()
+
         query_parameters = {
             "keep_types": [question_type],
```
Comment (alexgarel) [inline robotoff/app/api.py:1161]: Replace this by:
```suggestion
            "keep_types": [question_type] if question_type else None,
```

Comment (alexgarel) [reply robotoff/app/api.py:1161]: That said, I don't think using this API without question_type as any sense, so I would better make question_type mandatory.

File: robotoff/app/api.py [hunk 2]
```diff
@@ -1150,13 +1150,19 @@ def on_get(self, req: falcon.Request, resp: falcon.Response):
         count: int = req.get_param_as_int("count", min_value=1, default=25)
         question_type: str = req.get_param("type")
         value_tag: str = req.get_param("value_tag")
+        country: Optional[str] = req.get_param("country")
         server_domain: Optional[str] = req.get_param("server_domain")
 
+        import pdb
+
+        pdb.set_trace()
+
```
Comment (raphael0202) [inline robotoff/app/api.py:1159]: Can you remove this debugging code?

File: robotoff/app/core.py [hunk 3]
```diff
@@ -100,6 +100,8 @@ def get_insights(
     if keep_types:
         where_clauses.append(ProductInsight.type.in_(keep_types))
 
+    # import pdb; pdb.set_trace()
+
```
Comment (raphael0202) [inline robotoff/app/core.py:104]: Same comment as above about debugging

File: tests/integration/test_api.py [hunk 4]
```diff
@@ -655,6 +655,26 @@ def test_get_unanswered_questions_api(client):
     assert data["status"] == "found"
 
 
+def test_get_unanswered_questions_api_with_country_filter(client):
+    ProductInsight.delete().execute()  # remove default sample
+
+    # test for filter with "country"
+
+    d = ProductInsightFactory(
```
Comment (raphael0202) [inline tests/integration/test_api.py:663]: You don't have to assign the product to `d` variable as you're not using it (local variable 'd' is assigned to but never used)

**Review by raphael0202 — CHANGES_REQUESTED** [review]

File: tests/integration/test_core_integration.py [hunk 5]
```diff
@@ -172,30 +172,33 @@ def test_get_unanswered_questions_list():
...
     insight_data5 = get_insights(keep_types=["category"])
     insight_data_items5 = [item.to_dict() for item in insight_data5]
     assert len(insight_data_items5) == 2
 
+    product6 = ProductInsightFactory(value_tag="en:raisins", countries="en:india")
+    insight_data6 = get_insights(country="en:india")
+    insight_data_items6 = [item.to_dict() for item in insight_data6]
+    assert insight_data_items6[0]["id"] == product6.id
+    assert insight_data_items6[0]["value_tag"] == "en:raisins"
+    assert insight_data_items6[0]["countries"] == "en:india"
```
Comment (alexgarel) [inline tests/integration/test_core_integration.py:200]: you should also test you have only 1 product

```suggestion
    assert len(insight_data_items6) == 1
```

File: tests/integration/test_api.py [hunk 6]
```diff
@@ -655,6 +655,26 @@ def test_get_unanswered_questions_api(client):
...
 
 
+def test_get_unanswered_questions_api_with_country_filter(client):
+    ProductInsight.delete().execute()  # remove default sample
+
+    # test for filter with "country"
+
+    ProductInsightFactory(
+        type="location", value_tag="en:dates", barcode="032", countries=["en:india"]
+    )
```
Comment (alexgarel) [inline tests/integration/test_api.py:665]: Adding only one product, you don't really test the filter is effective.

To test that you should add another product in another country
```suggestion
    ProductInsightFactory(
        type="location", value_tag="en:dates", barcode="033", countries=["en:france"]
    )
```

**Review by alexgarel — APPROVED** [review]
We need just a small improvement on tests.

I validate the PR but please, apply suggestions before merging, if you agree.

**Pull request merged**